busybox/22-ifconfig-ipv6.patch

```diff
@@ -0,0 +1,151 @@
+
+diff --git a/networking/ifconfig.c b/networking/ifconfig.c
+index 0a91e71..9c22c61 100644
+--- a/networking/ifconfig.c
++++ b/networking/ifconfig.c
+@@ -100,6 +100,7 @@
+ #include "libbb.h"
+ #include "inet_common.h"
+ #include <net/if.h>
++#include <net/if6.h>
+ #include <net/if_arp.h>
+ #include <netinet/in.h>
+ #ifdef HAVE_NET_ETHERNET_H
+@@ -129,14 +130,6 @@
+ # define IFF_DYNAMIC     0x8000	/* dialup device with changing addresses */
+ #endif
+ 
+-#if ENABLE_FEATURE_IPV6
+-struct in6_ifreq {
+-	struct in6_addr ifr6_addr;
+-	uint32_t ifr6_prefixlen;
+-	int ifr6_ifindex;
+-};
+-#endif
+-
+ /*
+  * Here are the bit masks for the "flags" member of struct options below.
+  * N_ signifies no arg prefix; M_ signifies arg prefixed by '-'.
+@@ -272,8 +265,8 @@ static const struct arg1opt Arg1Opt[] = {
+ 	{ "SIFMAP",     SIOCSIFMAP,     ifreq_offsetof(ifr_map.irq) },
+ #endif
+ #if ENABLE_FEATURE_IPV6
+-	{ "SIFADDR",    SIOCSIFADDR,    ifreq_offsetof(ifr_addr) }, /* IPv6 version ignores the offset */
+-	{ "DIFADDR",    SIOCDIFADDR,    ifreq_offsetof(ifr_addr) }, /* IPv6 version ignores the offset */
++	{ "SIFADDR",    SIOCAIFADDR_IN6,    ifreq_offsetof(ifr_addr) }, /* IPv6 version ignores the offset */
++	{ "DIFADDR",    SIOCDIFADDR_IN6,    ifreq_offsetof(ifr_addr) }, /* IPv6 version ignores the offset */
+ #endif
+ 	/* Last entry is for unmatched (assumed to be hostname/address) arg. */
+ 	{ "SIFADDR",    SIOCSIFADDR,    ifreq_offsetof(ifr_addr) },
+@@ -453,15 +446,13 @@ int ifconfig_main(int argc UNUSED_PARAM, char **argv)
+ 							}
+ 							if (lsa->u.sa.sa_family == AF_INET6) {
+ 								int sockfd6;
+-								struct in6_ifreq ifr6;
++								struct in6_aliasreq ifr6;
+ 
+ 								sockfd6 = xsocket(AF_INET6, SOCK_DGRAM, 0);
+-								xioctl(sockfd6, SIOCGIFINDEX, &ifr);
+-								ifr6.ifr6_ifindex = ifr.ifr_ifindex;
+-								ifr6.ifr6_prefixlen = prefix_len;
+-								memcpy(&ifr6.ifr6_addr,
+-										&lsa->u.sin6.sin6_addr,
+-										sizeof(struct in6_addr));
++								strncpy_IFNAMSIZ(ifr6.ifra_name, ifr.ifr_name);
++								memcpy(&ifr6.ifrau_addr,
++										&lsa->u.sin6,
++										sizeof(struct sockaddr_in6));
+ 								ioctl_or_perror_and_die(sockfd6, a1op->selector, &ifr6, "SIOC%s", a1op->name);
+ 								if (ENABLE_FEATURE_CLEAN_UP)
+ 									free(lsa);
+diff --git a/networking/interface.c b/networking/interface.c
+index 228ccf7..0d6c19c 100644
+--- a/networking/interface.c
++++ b/networking/interface.c
+@@ -33,6 +33,7 @@
+ 
+ #include "libbb.h"
+ #include "inet_common.h"
++#include <ifaddrs.h>
+ #include <net/if.h>
+ #include <net/if_arp.h>
+ #ifdef HAVE_NET_ETHERNET_H
+@@ -52,7 +53,6 @@
+ #endif
+ 
+ #define _PATH_PROCNET_DEV               "/proc/net/dev"
+-#define _PATH_PROCNET_IFINET6           "/proc/net/if_inet6"
+ 
+ #ifdef HAVE_AFINET6
+ # ifndef _LINUX_IN6_H
+@@ -841,6 +841,7 @@ static void print_bytes_scaled(unsigned long long ull, const char *end)
+ 
+ 
+ #ifdef HAVE_AFINET6
++#if 0
```

Comment:
maaybe we could use eg. `!defined(__phoenix__)` to highlight that this is our custom implementation (while viewing code after patching)?